code/game/machinery/autodoc_excelsior.dm

```diff
@@ -0,0 +1,251 @@
+/obj/machinery/excelsior_autodoc
+	name = "Excelsior Autodoc"
+	desc = "Medical care for everybody, free, and may no one be left behind!"
+	icon = 'icons/obj/machines/excelsior/autodoc.dmi'
+	icon_state = "base"
+	density = TRUE
+	anchored = TRUE
+	circuit = /obj/item/electronics/circuitboard/excelsior_autodoc
+	use_power = IDLE_POWER_USE
+	idle_power_usage = 60
+	active_power_usage = 10000
+	var/mob/living/carbon/occupant
+	var/datum/autodoc/autodoc_processor
+	var/cover_closed = FALSE
+	var/cover_locked = FALSE
+	var/cover_moving = FALSE
+	var/image/screen_state = null
+	var/image/cover_state = null
+
+/obj/machinery/excelsior_autodoc/New()
+	. = ..()
+	autodoc_processor = new/datum/autodoc()
+	autodoc_processor.holder = src
+	autodoc_processor.damage_heal_amount = 20
+	update_icon()
+	
+
+/obj/machinery/excelsior_autodoc/Destroy()
+	if(occupant)
+		qdel(occupant)
+	return ..()
+
+/obj/machinery/excelsior_autodoc/relaymove(mob/user)
+	if (user.stat)
+		return
+	src.go_out()
+	return
+
+/obj/machinery/excelsior_autodoc/attackby(obj/item/I, mob/living/user)
+	if(default_deconstruction(I, user))
+		return
+	if(default_part_replacement(I, user))
+		return
+	..()
+
+/obj/machinery/excelsior_autodoc/verb/eject()
+	set src in view(1)
+	set category = "Object"
+	set name = "Eject Occupant"
+
+	if (usr.incapacitated())
+		return
+	src.go_out()
+	add_fingerprint(usr)
+	return
+
+/obj/machinery/excelsior_autodoc/verb/move_inside()
+	set src in view(1)
+	set category = "Object"
+	set name = "Enter Autodoc"
+
+	if(usr.stat)
+		return
+	if(src.occupant)
+		to_chat(usr, SPAN_WARNING("The autodoc is already occupied!"))
+		return
+	if(usr.abiotic())
+		to_chat(usr, SPAN_WARNING("The subject cannot have abiotic items on."))
+		return
+	set_occupant(usr)
+	src.add_fingerprint(usr)
+	return
+
+/obj/machinery/excelsior_autodoc/proc/go_out()
+	if (!occupant || cover_locked)
+		return
+	if(autodoc_processor.active)
+		to_chat(usr, SPAN_WARNING("Comrade, autodoc is locked down! Wait until all operations is done."))
+		return
+	if(cover_closed)
+		open_cover()
+
+	for(var/obj/O in src)
+		O.forceMove(loc)
+	occupant.forceMove(loc)
+	occupant.reset_view()
+	occupant.unset_machine()
+	occupant = null
+	autodoc_processor.set_patient(null)
+	update_use_power(1)
+	update_icon()
+
+/obj/machinery/excelsior_autodoc/proc/set_occupant(var/mob/living/L)
+	L.forceMove(src)
+	src.occupant = L
+	src.add_fingerprint(usr)
+	autodoc_processor.set_patient(L)
+	update_use_power(2)
+	L.set_machine(src)
+	cover_state = image(icon, "opened")
+	cover_state.layer = 4.5
+
+	if(!is_excelsior(L))
+		cover_locked = TRUE
+		close_cover()
+		sleep(20)
+		to_chat(L, SPAN_DANGER("Autodoc implanting you with something!"))
+		sleep(40)
+		var/obj/item/implant/excelsior/implant = new(L)
+		implant.install(L, BP_HEAD)
+		var/datum/faction/F = get_faction_by_id(FACTION_EXCELSIOR)
```

Comment:
leaving the balance issues of this aside, this code doesn't check for a cruciform and sets the faction anyways, which is problematic.

Reply:
Neither does regular implanter, cruciform rejects it anyway. Following code doesn't force faction to player, it triggers excelsior timer, as it should be, when someone implanted.

Reply:
ah, that's correct. Instead, it'll just runtime from implant.install's user not being defined and doesn't handle implant.install returning null instead of TRUE(can happen from cruciform or not having a head).

Reply:
Bullshit. I've tested on cruciformed and beheaded mobs, no runtimes popped up.

Reply:
alright, I really am off my game atm, no target is indeed handled.